contracts/StakeManager.sol

```diff
@@ -0,0 +1,165 @@
+pragma solidity ^0.5.16;
+pragma experimental ABIEncoderV2;
+
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+import "./interfaces/IRelayHub.sol";
+
+contract StakeManager {
+
+    /// Emitted when a stake or unstakeDelay are initialized or increased
+    event StakeAdded(
+        address indexed registree,
+        address indexed owner,
+        uint256 stake,
+        uint256 unstakeDelay
+    );
+
+    /// Emitted once a stake is scheduled for withdrawal
+    event StakeUnlocked(
+        address indexed registree,
+        address indexed owner,
+        uint256 withdrawBlock
+    );
+
+    /// Emitted when owner withdraws registree funds
+    event StakeWithdrawn(
+        address indexed registree,
+        address indexed owner,
+        uint256 amount
+    );
+
+    /// Emitted when an authorized Relay Hub penalizes a registree
+    event StakePenalized(
+        address indexed registree,
+        address indexed beneficiary,
+        uint256 reward
+    );
+
+    event HubAuthorized(
+        address indexed registree,
+        address indexed relayHub
+    );
+
+    event HubUnauthorized(
+        address indexed registree,
+        address indexed relayHub,
+        uint256 removalBlock
+    );
+
+    /// @param stake - amount of ether staked for this relay
+    /// @param unstakeDelay - number of blocks to elapse before the owner can retrieve the stake after calling 'unlock'
+    /// @param withdrawBlock - first block number 'withdraw' will be callable, or zero if the unlock has not been called
+    /// @param owner - address that receives revenue and manages registree's stake
+    struct StakeInfo {
+        uint256 stake;
+        uint256 unstakeDelay;
+        uint256 withdrawBlock;
+        address payable owner;
+    }
+
+    struct RelayHubInfo {
+        uint256 removalBlock;
+    }
+
+    /// maps owners to their registrees
+    mapping(address => address) public registrees;
```

Comment:
this means a single owner can haev a single registree. That's a problem.
what can I do to have multiple (separate) relays, and still use the same owner address ?
(each relay can have multiple addresses, but that's a separate issue, unrelated to SM)

Reply:
Why do you want one owner to have multiple registrees, again?

Reply:
I can have multiple relays. Each relay appear once in the list a client use, regardless of how many addresses it has. I want to have multiple relays - and to do so I have to stake more than once.

Reply:
No. Client uses list of relays. This is controlled by Relay Hub.
Owner puts a stake for "registree" address. This address can call "registerNewRelay". This is what gives him this name